src/hotspot/share/services/mallocTracker.inline.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023 SAP SE. All rights reserved.
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_SERVICES_MALLOCTRACKER_INLINE_HPP
+#define SHARE_SERVICES_MALLOCTRACKER_INLINE_HPP
+
+#include "services/mallocLimit.hpp"
+#include "services/mallocTracker.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/vmError.hpp"
+
+static inline bool suppress_limit_handling() {
```

Comment:
> Why did we bother to wrap `VMError::is_error_reported()` into `suppress_limit_handling()`?
> 

Because during error handling, code may malloc() too (bad practice, but it can happen). If it does, I don't want circular assertions to fire; I want a clean, complete hs-err file.

> Are you anticipating more exclusions here in the future?

None I can think of.

Reply:
That explains why we want to call `VMError::is_error_reported()` sure, but not why we wrapped it in `suppress_limit_handling()` SPI, unless I am missing something?

Reply:
Oh okay. This was mostly a documentation thing. I can remove the wrapper.